python/arcticdb/arctic.py

```diff
@@ -98,6 +98,8 @@ def __init__(self, uri: str, encoding_version: EncodingVersion = EncodingVersion
         self._library_manager = LibraryManager(self._library_adapter.config_library)
         self._uri = uri
 
+        self._library_adapter.check_storage_is_accessible()
```

Comment:
I think there should be a parameter in the URI that optionally disables this behaviour

Reply:
We may warn incorrectly if we can't `HEAD` the root path but can access other paths.

I also don't trust all S3 implementations to implement this correctly and uniformly - it wouldn't shock me if some implementations 404 HeadBucket operations but other ops work.

As a result I really think we should merge this with way to disable.